packages/material/src/complex/table-array.control.tsx

```diff
@@ -41,117 +55,325 @@ export const tableArrayTester: RankedTester = rankWith(10, and(
-                .map(prop => <TableCell key={prop}>{prop}</TableCell>)
-                .value()
-            }
+              <TableCell padding='checkbox' style={{width: '1em'}}>
+                <Checkbox
+                  indeterminate={numSelected > 0 && numSelected < rowCount}
+                  checked={numSelected === rowCount}
+                  onChange={handleSelectAllClick}
+                />
+              </TableCell>
+              {generateCells(TableHeaderCell, resolvedSchema, path)}
             </TableRow>
           </TableHead>
           <TableBody>
-          {
-            (!data || !Array.isArray(data) || data.length === 0) ?
-              <TableRow><TableCell>No data</TableCell></TableRow> : data.map((_child, index) => {
-              const childPath = Paths.compose(path, index + '');
-
-              return (
-                <TableRow key={childPath}>
-                  {
-                    _.chain(resolvedSchema.properties)
-                      .keys()
-                      .filter(prop => resolvedSchema.properties[prop].type !== 'array')
-                      .map((prop, idx) => {
-                        return (
-                          <TableCell key={Paths.compose(childPath, idx.toString())}>
-                            <DispatchField
-                              schema={resolvedSchema}
-                              uischema={createControlElement(prop)}
-                              path={childPath}
-                            />
-                          </TableCell>
-                        );
-                      })
-                      .value()
-                  }
-                </TableRow>
-              );
-            })
-          }
+            {isEmptyTable ? <EmptyTable/> : <TableWithContent {...tableProps}/>}
           </TableBody>
         </Table>
+        <Dialog
+          open={this.state.openConfirmDelete}
+          keepMounted
+          onClose={handleClose}
+          aria-labelledby='alert-dialog-confirmdelete-title'
+          aria-describedby='alert-dialog-confirmdelete-description'
+        >
+          <DialogTitle id='alert-dialog-confirmdelete-title'>
+            {'Confirm Deletion'}
+          </DialogTitle>
+          <DialogContent>
+            <DialogContentText id='alert-dialog-confirmdelete-description'>
+              Are you sure you want to delete the {this.state.selected.length} selected objects?
+            </DialogContentText>
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={handleClose} color='primary'>
+              No
+            </Button>
+            <Button onClick={removeItems} color='primary'>
+              Yes
+            </Button>
+          </DialogActions>
+        </Dialog>
       </div>
     );
   }
 }
+export interface TableProps extends ControlProps {
+  childErrors: ErrorObject[];
+}
+// FIXME copied from validation.ts
+const errorAt = instancePath => (state): ErrorObject[] => {
```

Comment:
Doesn't seem to be an exact copy. Why the difference? If we need this, then we should provide this extractor alongside the regular `errorAt` extractor?